### ISSUE TITLE
"Increase list level" action inserts keyboard key if the list indent is invalid

### ISSUE THREAD

**Issue by vjpr** [vsch/idea-multimarkdown#470]
```
- foo
<press-shortcut>- bar

- foo
  - bar

- foo
  <press-shortcut-again>- bar

// E.g. if we bind the shortcut to `alt + .` it will insert a special character `≥`.
- foo
  ≥-bar
```

First of all, I would expect it to do nothing rather than inserting the default shortcut.

Secondly, I think that it should allow you to make invalid indentations, otherwise it feels like Visual Basic popping up a modal dialog when you make a syntax error 😵 . All other editors don't prevent me from writing invalid code - as sometimes you need to as part of refactoring, etc.

I also think that when there is an empty list, the tab key should behave like this action (see https://github.com/vsch/idea-multimarkdown/issues/469).

**Comment by vsch** [comment]
@vjpr, that is IDE behaviour, not the plugin. When there are multiple actions assigned to the same shortcut, the IDE will apply a contextual shortcut if it is available. When it is not available, the next available action assigned to the shortcut will be invoked.

The plugin has no control over this. In the use case you present, the list indent action is disabled because there is no next list level. This causes your default action to be executed.

**Comment by vjpr** [comment]
> the list indent action is disabled because there is no next list level.

I am suggesting that it should not be disabled because there is no next list level. It should allow me to create invalid indentation.

This is how most editors work. This should generate a "lint" error, but I should still be able to perform that action.

Otherwise it is very restrictive and feels like a WYSIWIG editor instead of plain-text, which is the spirit of markdown.

**Comment by vsch** [comment]
@vjpr, the indent action is context sensitive and not just "insert spaces". It will "indent" the item and its sub-items, if it has any, indent action on item 2:

```markdown
1. item 1
2. item 2
   * sub item
     * sub-sub item
```

Will result in:

```markdown
1. item 1
   1. item 2
      * sub item
        * sub-sub item
```

If you want plain text editing behaviour then you should use tab or spaces. I will add config option for number of spaces for a tab then you can set it to 2 and get what you want. However, setting smart tabs, which will use context sensitive behaviour will result in the "indent list item action" when appropriate or simple tab when not. So you will get the above indentation behaviour when the caret is after the list item marker.

Markdown has complex context, especially when it is being edited getting the AST and text can be out of sync. Trying to do the right thing as far as being intuitive is very difficult with lots of unexpected gotchas.

This is the reason why I have been hesitant to put in "smart" tab behaviour. There is a lot of trial and error with tons of special conditions leading to very long stabilization periods that the users hate.

**Comment by vjpr** [comment]
> It will "indent" the item and its sub-items

This is cool, I didn't realise it did this and would be very useful.

> So you will get the above indentation behaviour when the caret is after the list item marker.

This works for me.

**Comment by vsch** [comment]
Addressed by #469 to use tab/backtab for indent/unindent
